doc/contributions/introduction.md

```diff
@@ -0,0 +1,80 @@
+# Introduction
+
+## Git and Github
+
+We recommend you to get familiar with the
+[git](https://guides.github.com/introduction/git-handbook) and
+[Github](https://guides.github.com) workflows before posting any changes.
+
+### Set up in a nut shell
+
+The following steps describe the process in a nut shell to provide you a basic
+template:
+
+* Create an account on [GitHub](https://github.com)
+* Fork the [shadow-utils repository](https://github.com/shadow-maint/shadow)
+* Clone the SSSD repository
+
+```
+git clone git@github.com:shadow-maint/shadow.git
+```
+
+* Add your fork as an extra remote
+
+```
+git remote add $ghusername git@github.com:$ghusername/shadow.git
+```
+
+* Setup your name contact e-mail that you want to use for the development
+
+```
+git config user.name "John Smith"
+git config user.email "john.smith@home.com"
+```
+
+**Note**: this will setup the user information only for this repository. You
+can also add `--global` switch to the `git config` command to setup these
+options globally and thus making them available in every git repository.
+
+* Create a working branch
+
+```
+git checkout -b my-changes
+```
+
+* Commit changes
+
+```
+vim change-what-you-need
+git commit -s
```

Comment:
Best to mention here to give a clear commit message.

Perhaps point to '2) Describe your changes' in https://www.kernel.org/doc/html/v4.14/process/submitting-patches.html , although if you have something shorter to point to that would be good.

Reply:
For the moment I referenced the section that you mentioned. We can improve it in the future (once I follow the process several times).